mlkem/ntt.c

```diff
@@ -74,49 +74,95 @@ const int16_t zetas[128] = {
  **************************************************/
 #if !defined(MLKEM_USE_NATIVE_NTT)
 
-// Check that the specific bound for the reference NTT implies
-// the bound required by the C<->Native interface.
-#define NTT_BOUND_REF (5 * MLKEM_Q)
-STATIC_ASSERT(NTT_BOUND_REF <= NTT_BOUND, ntt_ref_bound)
+// Compute a block CT butterflies with a fixed twiddle factor
+STATIC_TESTABLE
+// clang-format off
+void ntt_butterfly_block(int16_t *r, int16_t root, int start, int len, int bound)
+  REQUIRES(0 <= start && start < MLKEM_N && 1 <= len && len <= MLKEM_N / 2 && start + 2 * len <= MLKEM_N)
+  REQUIRES(0 <= bound && bound < INT16_MAX - MLKEM_Q)
+  REQUIRES(IS_FRESH(r, sizeof(int16_t) * MLKEM_N))
+  REQUIRES(SCALAR_ABS_BOUND(root, HALF_Q - 1))
+  REQUIRES(ARRAY_ABS_BOUND(r, 0, start - 1, bound + MLKEM_Q))
+  REQUIRES(ARRAY_ABS_BOUND(r, start, MLKEM_N - 1, bound))
+  ASSIGNS(OBJECT_UPTO(r, sizeof(int16_t) * MLKEM_N))
+  ENSURES(ARRAY_ABS_BOUND(r, 0, start + 2*len - 1, bound + MLKEM_Q))
```

Comment:
and similarly, explain what's going on here.. something like:

// After a new block has been butterflied, the set of modified coefficients with increased bound expands to cover coefficients 0 through start+2*len-1

Reply:
As below -- I'm not sure such comments are a net benefit because they distract further from the actual code, but I could be convinced otherwise.

Reply:
existing comment at the top is sufficient i believe - better extend that description rather than adding more inline comments.